beluga/include/beluga/algorithm/unscented_transform.hpp

```diff
@@ -58,46 +63,51 @@ auto unscented_transform(
       "Mean and covariance scalar types differ.");
   static_assert(std::is_floating_point_v<Scalar>);
   static_assert(MeanDerived::ColsAtCompileTime == 1, "Mean should be a column vector.");
-
+  static_assert(
+      CovarianceDerived::ColsAtCompileTime == CovarianceDerived::RowsAtCompileTime, "Covariance matrix is not square.");
   constexpr int kNin = MeanDerived::RowsAtCompileTime;
+  static_assert(kNin >= 1, "Input dimension should be greater than zero.");
+  static_assert(CovarianceDerived::ColsAtCompileTime == kNin);
   const double k = kappa.value_or(std::max(kNin - 3, 0));
   assert(k >= 0);
   const Scalar w0 = static_cast<Scalar>(k) / static_cast<Scalar>(kNin + k);
   std::vector<Eigen::Vector<Scalar, kNin>> sigma_points;
   std::vector<Scalar> weights;
   sigma_points.reserve(2 * kNin + 1);
-  weights.reserve(2 * kNin + 1);
+  weights.resize(2 * kNin + 1);
   sigma_points.emplace_back(mean);
-  weights.emplace_back(w0);
+  weights[0] = w0;
   const Scalar wn = 1. / (2. * (kNin + k));
 
   // Compute the Cholesky decomposition of the covariance matrix
-  const Eigen::Matrix<Scalar, kNin, kNin> chol_cov = covariance.llt().matrixL();
+  const auto llt = covariance.llt();
+  assert(llt.info() == Eigen::ComputationInfo::Success);
+  const Eigen::Matrix<Scalar, kNin, kNin> l_matrix = llt.matrixL();
+  // Precompute sqrt{n + kaappa} * L for speed sake.
+  const Eigen::Matrix<Scalar, kNin, kNin> aux = std::sqrt(kNin + k) * l_matrix;
```

Comment:
@serraramiro1 nit: 
```suggestion
  const Eigen::Matrix<Scalar, kNin, kNin> scaled_l_matrix = std::sqrt(kNin + k) * l_matrix;
```
? Or anything a bit more specific than `aux`.